### ISSUE TITLE
[Bug]: 5.71.0 Not Available via registry.terraform.io

### ISSUE THREAD

**Issue by bjf-rga** [hashicorp/terraform-provider-aws#39694]
### Terraform Core Version

n/a

### AWS Provider Version

5.71.0

### Affected Resource(s)

All

### Expected Behavior

aws provider 5.71.0 downloads

### Actual Behavior

```
- Finding hashicorp/aws versions matching "5.71.0"...
╷
│ Error: Failed to query available provider packages
│ 
│ Could not retrieve the list of available versions for provider hashicorp/aws: no available releases match the given constraints 5.71.0
╵
```

### Relevant Error/Panic Output Snippet

```shell
$ curl https://registry.terraform.io/v1/providers/hashicorp/aws/versions | jq .versions[].version | sort -V | tail -2
"5.69.0"
"5.70.0"
```

### Terraform Configuration Files

```
terraform {
  required_version = "= 1.9.7"
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "= 5.71.0"
    }
  }
}
```

### Steps to Reproduce

```
terraform init -upgrade -reconfigure
```

### Debug Output

n/a

### Panic Output

n/a

### References

While 5.71.0 is listed as the latest release here, it's not available via registry.terraform.io

### Would you like to implement a fix?

None

**Comment by jahaniam** [comment]
https://github.com/hashicorp/terraform-provider-aws/issues/39681#issuecomment-2407642106

Looks like the package is retracted but the registry is still pointing to the new version.

This fails our pipelines.

**Comment by rhowe** [comment]
5.71.0 is listed in the versions output, but there are no platforms specified for it.

Compare

    curl -fs https://registry.terraform.io/v1/providers/hashicorp/aws/versions | jq '.versions[]|select(.version == "5.70.0")|.platforms'

vs

    curl -fs https://registry.terraform.io/v1/providers/hashicorp/aws/versions | jq '.versions[]|select(.version == "5.71.0")|.platforms'

**Comment by bjf-rga** [comment]
While the issue @rhowe mentions persists (no platforms returned), I can confirm that deployments pinned to `5.71.0` are now operational, at least for me.

**Comment by jasonm3-netapp** [comment]
I know this is referencing 5.71.0, but we're also having the issue with 5.65.0, which is where we're pinned.

**Comment by justinretzolk** [comment]
Hey all 👋 Thank you for taking the time to raise this! `5.71.0` was indeed removed due to an issue with the release, and will not be re-released. All changes that were included in that release will be included in the upcoming release (`5.72.0`) on Thursday. We've updated the [changelog](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#5720-unreleased) to mention this.

There aren't logs for me to go off of, but for those not explicitly attempting to pin to provider version `5.71.0`, I suspect a similar thing that I mentioned in #39700 where caching may be causing Terraform to attempt to re-download the now-removed version.

Since there's no further action for the AWS Provider team to take, I'm going to close this issue. If you encounter unexpected behavior in the future, please do let us know.
